homeassistant/components/edl21/sensor.py

```diff
@@ -36,6 +42,183 @@
     },
 )
 
+# OBIS format: A-B:C.D.E*F
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    # A=1: Electricity
+    # C=0: General purpose objects
+    # D=0: Free ID-numbers for utilities
+    SensorEntityDescription(key="1-0:0.0.9*255", name="Electricity ID"),
+    # D=2: Program entries
+    SensorEntityDescription(
+        key="1-0:0.2.0*0",
+        name="Configuration program version number",
+    ),
+    SensorEntityDescription(key="1-0:0.2.0*1", name="Firmware version number"),
+    # C=1: Active power +
+    # D=8: Time integral 1
+    # E=0: Total
+    SensorEntityDescription(
+        key="1-0:1.8.0*255",
+        name="Positive active energy total",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.ENERGY,
```

Comment:
When an entity sets a device class, it should not set an icon. However, the `EDL21Entity` sets an fixed entity.

Instead, remove the icon property method from `EDL21Entity` and set an icon on the EntityDescriptions that do not have a device class.